Sources/Application/TokenHolder.swift

```diff
@@ -3,16 +3,23 @@ import Networking
 
 enum TokenError: Error {
     case bearerNotPresent
+    case expired
 }
 
 class TokenHolder: AuthenticationProvider {
+    static let shared = TokenHolder()
```

Comment:
I've made a singleton for the TokenHolder as there should only ever be one place where we hold tokens in memory for the app. This avoids having to pass it around everywhere and makes it easily accessible from anywhere it's needed. Happy to revert if we think this is insecure or prone to other issues.